File: 1537-MaximumScoreAfterSplittingAString/1537-MaximumScoreAfterSplittingAString.py
# Last updated: 6/22/2026, 12:52:33 PM
class Solution:
    def maxScore(self, s: str) -> int:
        zero: int  = s[0] == '0'
        
        one: int = 0
        for i in s[1:]:
            if i == '1':
                one += 1

        max: int = one + zero

        for i in s[1:-1]:
            if i == '0':
                zero += 1
            else:
                one -= 1

            if max < zero + one:
                max = zero + one

        return max
        

File: 1537-MaximumScoreAfterSplittingAString/1537-MaximumScoreAfterSplittingAString.js
// Last updated: 6/22/2026, 12:52:46 PM
/**
 * @param {string} s
 * @return {number}
 */
var maxScore = function(s) {
    let zero = s[0] == '0';
    let one = 0;
    for(let i=1; i<s.length; i++){
        if(s[i] == '1') one++;
    }

    let max = zero+one;
    for(let i=1; i<s.length-1; i++){
        s[i] == '0' ? zero++ : one--;
        if(max < zero+one){
            max = zero + one;
        }
    }

    return max;
};